staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -66,8 +66,8 @@ type crdHandler struct {
 	groupDiscoveryHandler   *groupDiscoveryHandler
 
 	customStorageLock sync.Mutex
-	// customStorage contains a crdStorageMap
-	customStorage atomic.Value
+	// cachedCrdStorage contains a customresourcedefinition storage map
+	cachedCrdStorage map[types.UID]*crdInfo
```

Comment:
why was the atomic value not suited?

Reply:
We still need a Mutex to protect the crdStorageMap， does not release our lock burden.

Reply:
The atomic value protects the pointer to the map only. We do not protect the values as we don't have to if we only mutate a copy of it. That's what the old code did.